doc/guides/method.rdoc

```diff
@@ -0,0 +1,35 @@
+= Documenting a Method
+
+== Structure
+
+=== Synopsis
+
+Begin with a one-line synopsis that tells what the method does.  Example:
+  *  call-seq:
+  *    array <=> other_array -> -1, 0, or 1
+  *
+  *  Returns -1, 0, or 1 as +self+ is less than, equal to, or greater than +other_array+.
+
+== Multiple Signatures
+
+\[Is 'signatures' the correct term?]
+
+When a method has multiple signatures,
+document each signature separately, as if it were a distinct method.
+
+In <tt>array.c</tt>, for example, we have this:
+  *  call-seq:
+  *    array[index] -> object or nil
+  *    array[start, length] -> object or nil
+  *    array[range] -> object or nil
+  *    array.slice(index) -> object or nil
+  *    array.slice(start, length) -> object or nil
+  *    array.slice(range) -> object or nil
+
+Document each signature separately, beginning with a level-6 header:
```

Comment:
I prefer to only use headings in method documentation in exceptional circumstances, but we definitely shouldn't jump headings.  You should only use a level-6 heading if you are currently under a level-5 heading.